src/olympia/addons/utils.py

```diff
@@ -105,19 +105,26 @@ def verify_mozilla_trademark(name, user):
 
     def _check(name):
         name = name.lower()
```

Comment:
A little more cleaning than just `.lower()` would make it more robust- maybe collapse double spaces and line returns to a single space; trim trailing spaces and punctuation? e.g. `Splendid Add-on For  Firefox! ` seems like it should pass.